pyro/ops/einsum/util.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import, division, print_function
+
+EINSUM_SYMBOLS_BASE = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
+
+
+class Tensordot(object):
+    """
+    Creates a tensordot implementation from an einsum implementation.
+    """
+    def __init__(self, einsum):
+        self.einsum = einsum
+
+    # Copyright (c) 2014 Daniel Smith
+    # This function is copied and adapted from:
+    # https://github.com/dgasmith/opt_einsum/blob/a6dd686/opt_einsum/backends/torch.py
+    def __call__(self, x, y, axes=2):
```

Comment:
This method was moved nearly verbatim from torch_log.py; it has merely been wrapped in a class.